src/coreclr/jit/emit.h

```diff
@@ -3458,6 +3476,18 @@ inline unsigned emitter::emitGetInsCIargs(instrDesc* id)
 //
 emitAttr emitter::emitGetMemOpSize(instrDesc* id) const
 {
+    if (id->idIsEvexbContext())
+    {
+        // should have the assumption that Evex.b now stands for the embedded broadcast context.
+        switch (id->idIns())
+        {
+            case INS_addps:
+                return EA_4BYTE;
+
+            default:
+                break;
+        }
```

Comment:
`TupleType` might not be enough to tell the pointer size in this case. For example, both `addps` and `addpd` has a tupletype of `FULL`, but based on the input size, the resulting pointer size may be different. 

Since the key can be the input size, a suggestion is that make [GetInputSizeInBytes()](https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/emitxarch.cpp#L15434) to be a const method, such that this method can be called in [GetMemOpSize()](https://github.com/dotnet/runtime/blob/main/src/coreclr/jit/emit.h#L3459).

Reply:
Right.

I meant rather that we can make this more table driven based on the total combination of `TupleType`, `InputSize` and `EmitAttr`.